source/features/useful-not-found-page.tsx

```diff
@@ -36,55 +38,112 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
-	const parts = parseCurrentURL();
-	parts[2] = 'commits';
-	const url = '/' + parts.join('/');
-	if (await is404(location.origin + url)) {
+async function getLatestChangeToFile(): Promise<Record<string, any> | void> {
+	const url = new GitHubURL(location.href);
+	const currentBranch = url.branch;
+	const {filePath} = url;
+	if (!currentBranch || !filePath) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			See also the {getType()}’s <a href={url}>commit history</a>.
-		</p>,
-	);
+	const commitsResponseObject = await api.v4(`
+		repository() {
+			ref(qualifiedName: "${currentBranch}") {
+				target {
+					... on Commit {
+						history(first: 1, path: "${filePath}") {
+							nodes {
+								oid
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+	const commits =
+		commitsResponseObject.repository.ref.target.history.nodes[0];
```

Comment:
Disable Prettier entirely on this project if you haven't

```suggestion
	const commits = commitsResponseObject.repository.ref.target.history.nodes[0];
```